cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -74,6 +74,71 @@ var (
 // generateMountsMode* is called multiple times from initramfs until it
 // no longer generates more mount points and just returns an empty output.
 func generateMountsModeInstall(recoverySystem string) error {
+	var buf bytes.Buffer
+	if err := generateMountsCommonInstallRecover(recoverySystem, &buf); err != nil {
+		return err
+	}
+	if buf.Len() > 0 {
+		io.Copy(stdout, &buf)
+		return nil
+	}
+
+	// n+1: final step: write $(ubuntu_data)/var/lib/snapd/modeenv - this
+	//      is the tmpfs we just created above
+	modeEnv := &boot.Modeenv{
+		Mode:           "install",
+		RecoverySystem: recoverySystem,
+	}
+	if err := modeEnv.Write(filepath.Join(runMnt, "ubuntu-data", "system-data")); err != nil {
+		return err
+	}
+
+	// n+2: done, no output, no error indicates to initramfs we are done
+	//      with mounting stuff
+	return nil
+}
+
+func generateMountsModeRecover(recoverySystem string) error {
+	var buf bytes.Buffer
+	if err := generateMountsCommonInstallRecover(recoverySystem, &buf); err != nil {
+		return err
+	}
+	if buf.Len() > 0 {
+		io.Copy(stdout, &buf)
+		return nil
+	}
+
+	// n+1: mount ubuntu-data for recovery
+	recoverDataDir := filepath.Join(runMnt, "recover-ubuntu-data")
+	isRecoverDataMounted, err := osutilIsMounted(recoverDataDir)
+	if err != nil {
+		return err
+	}
+	if !isRecoverDataMounted {
+		// TODO:UC20: does the encrypted ubuntu-data need a different
```

Comment:
Using the same label for the filesystem created in the encrypted device should work fine.

Reply:
Luks partition has already been opened by the time this runs right?